src/main/java/org/ossgang/commons/observables/DispatchingObservable.java

```diff
@@ -102,14 +103,28 @@ protected void dispatchValue(T newValue, Predicate<Set<SubscriptionOption>> opti
     }
 
     protected void dispatchException(Throwable exception) {
-        listeners.keySet().forEach(l -> dispatch(l::onException, exception));
+        AtomicBoolean wasDispatched = new AtomicBoolean(false);
+        listeners.keySet().forEach(l -> {
+            dispatch(l::onException, exception);
+            wasDispatched.set(true);
+        });
+        if (!wasDispatched.get()) {
+            dispatchToUncaughtExceptionHandler(new UnhandledException(exception));
+        }
```

Comment:
In order to reduce code duplication (before it was just one line...), perhaps we should delegate here to
```java
dispatchException(exception, any -> true)
```
?

Reply:
Agree, it should have been like this anyway :wink: